display-dimensions/src/main/java/com/esri/arcgismaps/sample/displaydimensions/MainActivity.kt

```diff
@@ -0,0 +1,139 @@
+        activityMainBinding.optionsButton
+    }
+
+    // keep an instance of the MapView's dimension layer
+    private var dimensionLayer: DimensionLayer? = null
+
+    // track if the layer is enabled
+    private var isDimensionLayerEnabled: Boolean = true
+
+    // track if the custom definition is enabled
+    private var isDefinitionEnabled: Boolean = false
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        // authentication with an API key or named user is
+        // required to access basemaps and other location services
+        ArcGISEnvironment.apiKey = ApiKey.create(BuildConfig.API_KEY)
+        lifecycle.addObserver(mapView)
+
+        // check if the .mmpk file exits
+        val mmpkFile = File(provisionPath, getString(R.string.file_name))
+        if (!mmpkFile.exists()) return showError("Mobile map package file does not exist.")
+
+        // create and load a mobile map package
+        val mobileMapPackage = MobileMapPackage(mmpkFile.path)
+
+        lifecycleScope.launch {
+            // load the mobile map package
+            mobileMapPackage.load().getOrElse {
+                return@launch showError("Failed to load the mobile map package: ${it.message}")
+            }
+            // if the loaded mobile map package does not contain a map
+            if (mobileMapPackage.maps.isEmpty()) {
+                return@launch showError("Mobile map package does not contain a map")
+            }
+
+            // add the map from the mobile map package to the map view,
+            // and set a min scale to maintain dimension readability
+            mapView.map = mobileMapPackage.maps[0]
+            mapView.map?.minScale = 35000.0
+
+            // set the dimension layer within the map
+            dimensionLayer = mapView.map?.operationalLayers?.firstOrNull { layer ->
+                layer is DimensionLayer
+            } as DimensionLayer
+
+        }
+
+        optionsButton.setOnClickListener {
+            // inflate the dialog layout and get references to each of its components
+            val dialogBinding = DimensionsDialogLayoutBinding.inflate(layoutInflater)
+            val dimensionLayerSwitch = dialogBinding.dimensionLayerSwitch.apply {
+                isChecked = isDimensionLayerEnabled
+            }
+            val definitionSwitch = dialogBinding.definitionSwitch.apply {
+                isChecked = isDefinitionEnabled
+            }
+
+            // set up the dialog
+            AlertDialog.Builder(this).apply {
+                setView(dialogBinding.root)
+                setTitle("${getString(R.string.settings)}: ${dimensionLayer?.name}")
+
+                dimensionLayerSwitch.setOnCheckedChangeListener { _, isEnabled ->
+                    // set the visibility of the dimension layer
+                    dimensionLayer?.isVisible = isEnabled
+                    isDimensionLayerEnabled = isEnabled
```

Comment:
Could these just go in the apply blocks where you initialize `dimensionLayerSwitch`? Same for `definitionSwitch.setOnCheckedChangeListener`